### PR TITLE
build: kube-dns: remove out of date docs

### PR DESCRIPTION
**Pull request by philips**
This is doc seems out of place. It is next to the build system for this
addon and is out of date, with del tags blocking out parts. Lets delete
it and I will merge the good parts over to the admin guide:
http://kubernetes.io/docs/admin/dns/

cc @girishkalele

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32931)

### CODE REVIEW

**Comment by philips** [comment]
Moved to https://github.com/kubernetes/kubernetes.github.io/pull/1251

**Comment by girishkalele** [comment]
cc @thockin who (I think) originally wrote the build/kube-dns README

File: build/kube-dns/README.md [hunk 1]
```diff
@@ -1,257 +1,15 @@
...
+enable by default. The source code is in [cmd/kube-dns][kube-dns].
 
-## What things get DNS names?
-The only objects to which we are assigning DNS names are Services.  Every
-Kubernetes Service is assigned a virtual IP address which is stable as long as
-the Service exists (as compared to Pod IPs which can change over time due to
-crashes or scheduling changes).  This maps well to DNS, which has a long
-history of clients that, on purpose or on accident, do not respect DNS TTLs
-(see previous remark about Pod IPs changing).
+The [Kubernetes DNS Admin Guide][dns-admin] provides further details on this plugin.
```
Comment (thockin) [inline build/kube-dns/README.md]: No it doesn't.  It links back to this file.

Comment (thockin) [inline build/kube-dns/README.md]: oh, I see the other PR now :)

**Comment by thockin** [comment]
OK once other PR is in

**Comment by philips** [comment]
@thockin the other PR has landed so this can be merged.

**Comment by thockin** [comment]
LGTM - check CI error (verify something something)

**Comment by philips** [comment]
@thockin can you put retest-not-required-docs-only and lgtm on this. It looks like the tests are failing to bring up clusters.

**Comment by thockin** [comment]
Some md files are missing ga-beacon analytics link:
/go/src/k8s.io/kubernetes/build/kube-dns/README.md
Docs need munging. Please run hack/update-munge-docs.sh

**Comment by philips** [comment]
@thockin Rebased and fixed the beacon. `$ ./hack/verify-munge-docs.sh` said `Docs are properly munged.`. I don't know why e2e tests aren't passing but this doc change isn't the reason.

**Comment by thockin** [comment]
@k8s-bot cvm gke e2e test this

**Comment by thockin** [comment]
@k8s-bot cvm gke e2e test this

**Comment by thockin** [comment]
rebase and we can manually merge, sorry.

**Comment by philips** [comment]
@thockin rebased.

**Pull request merged**